File: examples/fp-verification-demo-go/src/tp_true_positives.go
package main

import (
	"database/sql"
	"fmt"
	"os/exec"
)

// TP-01: SQL 注入 — 拼接
func vulnerableQuery(db *sql.DB, userId string) (*sql.Row, error) {
	query := fmt.Sprintf("SELECT * FROM users WHERE id = %s", userId)
	return db.QueryRow(query), nil
}

// TP-02: 命令注入 — shell 拼接
func vulnerableExec(input string) (string, error) {
	out, err := exec.Command("sh", "-c", "echo "+input).Output()
	return string(out), err
}

File: examples/js-vuln-demo/src/tp_true_positives.js
// TP — True Positives: 真实漏洞
const { exec } = require('child_process');

// TP-01: SQL 注入 — 模板字符串拼接
function vulnerableQuery(db, userId) {
    return db.query(`SELECT * FROM users WHERE id = ${userId}`);
}

// TP-02: 命令注入 — shell 拼接
function vulnerableExec(input) {
    exec(`ping ${input}`, (err, stdout) => {});
}
